File: Tutorial5-3Dagger-Android-MutipleActivitiesAndFragments/src/main/java/com/example/tutorial5_3dagger_android_mutipleactivitiesandfragments/second/SecondActivityModule.java
package com.example.tutorial5_3dagger_android_mutipleactivitiesandfragments.second;

import android.content.Context;


import com.example.tutorial5_3dagger_android_mutipleactivitiesandfragments.di.ActivityScope;
import com.example.tutorial5_3dagger_android_mutipleactivitiesandfragments.second.SecondActivity;
import com.example.tutorial5_3dagger_android_mutipleactivitiesandfragments.model.ToastMaker;

import dagger.Binds;
import dagger.Module;
import dagger.Provides;

@Module
public abstract class SecondActivityModule {

    /*
    NOTE: This can only be SecondActivity(AndroidInjector<T>), Activity does not work
     */
    @Binds
    abstract Context bindContext(SecondActivity activity);

    @ActivityScope
    @Provides
    public static ToastMaker provideToastMaker(Context context) {
        return new ToastMaker(context);
    }
}

File: Tutorial5-3Dagger-Android-MultipleActivitiesAndFragments/src/main/java/com/example/tutorial5_3dagger_android_multipleactivitiesandfragments/second/SecondActivityModule.kt
package com.example.tutorial5_3dagger_android_multipleactivitiesandfragments.second

import android.content.Context
import com.example.tutorial5_3dagger_android_multipleactivitiesandfragments.di.scope.ActivityScope
import com.example.tutorial5_3dagger_android_multipleactivitiesandfragments.model.ToastMaker
import dagger.Binds
import dagger.Module
import dagger.Provides

@Module
abstract class SecondActivityModule {
    /*
    NOTE: This can only be SecondActivity(AndroidInjector<T>), Activity does not work
     */
    @Binds
    abstract fun bindContext(activity: SecondActivity): Context

    @Module
    companion object {

        @JvmStatic
        @ActivityScope
        @Provides
        fun provideToastMaker(context: Context): ToastMaker {
            return ToastMaker(context)
        }

    }
}